src/regional_settings.cpp

```diff
@@ -172,19 +172,26 @@ static void load_forest_mapgen_settings( const JsonObject &jo,
                 if( forest_biome_ter_keys.empty() ) {
                     forest_biome_jo.throw_error( "Biome is not associated with any terrains." );
                 }
-                std::string first_ter = forest_biome_ter_keys.get_string( 0 );
-                load_forest_biome( forest_biome_jo, forest_mapgen_settings.unfinalized_biomes[first_ter],
-                                   overlay );
-                for( size_t biome_ter_idx = 1; biome_ter_idx < forest_biome_ter_keys.size(); biome_ter_idx++ ) {
+                std::string default_ter_name = member.name();
+                auto default_ter_it = forest_mapgen_settings.unfinalized_biomes.find( default_ter_name );
+                if( default_ter_it == forest_mapgen_settings.unfinalized_biomes.end() ) {
+                    forest_biome_jo.throw_error( "Terrain " + default_ter_name +
+                                                 " is not defined, cannot be used for forest biome." );
+                }
```

Comment:
I think this error should only throw if `overlay` is `true`. At some point the biome has to be created after all.